AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Sync/MutationSync/OutgoingMutationQueue/OutgoingMutationQueue.swift

```diff
@@ -201,8 +201,10 @@ final class OutgoingMutationQueue: OutgoingMutationQueueBehavior {
         }
 
         Task {
+            let latestSyncMetadata = try? storageAdapter.queryMutationSyncMetadata(for: mutationEvent.modelId, modelName: mutationEvent.modelName)
             let syncMutationToCloudOperation = await SyncMutationToCloudOperation(
                 mutationEvent: mutationEvent,
+                latestSyncMetadata: latestSyncMetadata,
```

Comment:
I think the options are..
1. When the MutationEvent is queried from the MutationEvent table, we also do a second query to get from MutationSyncMetadata, and store the version in the MutationEvent (probably do a hard copy). MutationEvent gets passed to SyncMutationToCloudOperation. This option means SyncMutationToCloudOperation doesn't have to change. MutationEvent is a "snapshot" of the model, version, etc. We're just making sure as we dequeue, the "snapshots" we work off always have the latest version. This also means MutationEvent.version is not being deprecated, just rehydrated when syncing.
2. Hydrate the version before creating the SyncMutationToCloudOperation, timing wise is the same as option 1. SyncMutationToCloudOperation's logic should use `latestSyncMetadata.version` and not use mutationEvent.version. It is odd to have "version" properties, so deprecating MutationEvent.version makes sense. However there are other implications of deprecating MutationEvent.version (discussion in another thread).
3. Hydrate the version within the execution of SyncMutationToCloudOperation, this way we optimize on getting the latest version at all times. If we hydrate it before, option 1 and 2, we don't know the delta between that version and the time that the SyncMutationToCloudOperation is actually executing its `main()` function. 

I think option 3 will have the smallest time gap between sending the request and getting the latest version for that model

Reply:
I see your point. The option#3 is the most optimized one. 

Initially, I attempted to pass the async escape closure to `SyncMutationToCloudOperation` in order to retrieve the `latestSyncMetadata`. This would be executed at the very last moment when generating the API request for the mutation event. However, I observed that this operation involves a lock and spans across various code snippets. Introducing an async closure within a lock context would need many modifications. Consequently, I opted to take a step back and simplify the change by resolving the `latestSyncMetadata` separately and passing it in.